services/registry/middleware/bearer-auth.js

```diff
@@ -34,16 +23,34 @@ function createBearerAuthMW() {
         );
       }
 
+      // getting access to the redis doesn't get you the tokens.
+      const hash = crypto
+        .createHash('sha256')
+        .update(bearer + process.env.SESSION_SECRET)
+        .digest('base64');
+
+      const key = `token_${hash}`;
+      let data = await context.redis.getAsync(key);
+
+      // eslint-disable: no-empty
       try {
-        const user = await Token.lookupUser(bearer);
-        if (user) {
-          context.user = user;
+        data = JSON.parse(data);
+        context.user = data;
+      } catch {}
+
+      if (data === null) {
+        const [err, result] = await context.storageApi
+          .getToken(bearer)
```

Comment:
`getToken()` sends the token in a `{token}` header so sensitive data does not get logged. 🤫